File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/comment/direct/Java/Go/s456888268_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

var (
	mod    = int64(1e9 + 7)
	dx     = []int{-1, 0, 1, 0}
	dy     = []int{0, -1, 0, 1}
	dx8    = []int{-1, -1, -1, 0, 0, 1, 1, 1}
	dy8    = []int{-1, 0, 1, -1, 1, -1, 0, 1}
	inf    = int(^uint(0) >> 1)
	linf   = int64(^uint(0) >> 1)
	dinf   = float64(^uint(0) >> 1)
	eps    = 1e-10
	pi     = 3.14159265358979323846
	sb     strings.Builder
	reader *bufio.Reader
	writer *bufio.Writer
)

func main() {
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	solve()
}

func solve() {
	tmp := ns()
	s := []rune(tmp)
	n := len(s)
	ans := 0
	cnt := 0
	for i := 0; i < n; i++ {
		if s[i] == 'S' {
			cnt++
		} else {
			if cnt == 0 {
				continue
			}
			ans++
			cnt--
		}
	}
	fmt.Fprintf(writer, "%d\n", n-2*ans)
}

func reverse(ar []int) {
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func reverseL(ar []int64) {
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func reverseD(ar []float64) {
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func reverseR(ar []rune) {
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func getReverse(s string) string {
	c := []rune(s)
	reverseR(c)
	return string(c)
}

func reverseList[T any](ls []T) {
	length := len(ls)
	for i := 0; i < length/2; i++ {
		ls[i], ls[length-1-i] = ls[length-1-i], ls[i]
	}
}

func sbnl() {
	sb.WriteString("\n")
}

func lowerBound(ar []int, x int) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] < x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func upperBound(ar []int, x int) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] <= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rlowerBound(ar []int, x int) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] > x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rupperBound(ar []int, x int) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] >= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func lowerBoundL(ar []int64, x int64) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] < x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func upperBoundL(ar []int64, x int64) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] <= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rlowerBoundL(ar []int64, x int64) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] > x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rupperBoundL(ar []int64, x int64) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] >= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func lowerBoundD(ar []float64, x float64) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] < x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func upperBoundD(ar []float64, x float64) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] <= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rlowerBoundD(ar []float64, x float64) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] > x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rupperBoundD(ar []float64, x float64) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] >= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func lowerBoundR(ar []rune, x rune) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] < x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func upperBoundR(ar []rune, x rune) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] <= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rlowerBoundR(ar []rune, x rune) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] > x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rupperBoundR(ar []rune, x rune) int {
	l, r := -1, len(ar)
	for r-l > 1 {
		c := (l + r) / 2
		if ar[c] >= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func lowerBoundList[T comparable](ls []T, x T) int {
	l, r := -1, len(ls)
	for r-l > 1 {
		c := (l + r) / 2
		if ls[c] < x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func upperBoundList[T comparable](ls []T, x T) int {
	l, r := -1, len(ls)
	for r-l > 1 {
		c := (l + r) / 2
		if ls[c] <= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rlowerBoundList[T comparable](ls []T, x T) int {
	l, r := -1, len(ls)
	for r-l > 1 {
		c := (l + r) / 2
		if ls[c] > x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rupperBoundList[T comparable](ls []T, x T) int {
	l, r := -1, len(ls)
	for r-l > 1 {
		c := (l + r) / 2
		if ls[c] >= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func concat(x int, arr []int) []int {
	ret := make([]int, len(arr)+1)
	copy(ret[1:], arr)
	ret[0] = x
	return ret
}

func concatArr(arr []int, x int) []int {
	ret := make([]int, len(arr)+1)
	copy(ret[:len(arr)], arr)
	ret[len(arr)] = x
	return ret
}

func concatL(x int64, arr []int64) []int64 {
	ret := make([]int64, len(arr)+1)
	copy(ret[1:], arr)
	ret[0] = x
	return ret
}

func concatArrL(arr []int64, x int64) []int64 {
	ret := make([]int64, len(arr)+1)
	copy(ret[:len(arr)], arr)
	ret[len(arr)] = x
	return ret
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func max3(a, b, c int) int {
	return max(max(a, b), c)
}

func min3(a, b, c int) int {
	return min(min(a, b), c)
}

func maxL(a, b int64) int64 {
	if a > b {
		return a
	}
	return b
}

func minL(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

func max3L(a, b, c int64) int64 {
	return maxL(maxL(a, b), c)
}

func min3L(a, b, c int64) int64 {
	return minL(minL(a, b), c)
}

func maxD(a, b float64) float64 {
	if a > b {
		return a
	}
	return b
}

func minD(a, b float64) float64 {
	if a < b {
		return a
	}
	return b
}

func max3D(a, b, c float64) float64 {
	return maxD(maxD(a, b), c)
}

func min3D(a, b, c float64) float64 {
	return minD(minD(a, b), c)
}

func sortInt(ar []int) {
	sort.Ints(ar)
}

func sortInt64(ar []int64) {
	sort.Slice(ar, func(i, j int) bool { return ar[i] < ar[j] })
}

func sortFloat64(ar []float64) {
	sort.Float64s(ar)
}

func sortRune(ar []rune) {
	sort.Slice(ar, func(i, j int) bool { return ar[i] < ar[j] })
}

func rSortInt(ar []int) {
	sort.Ints(ar)
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func rSortInt64(ar []int64) {
	sort.Slice(ar, func(i, j int) bool { return ar[i] < ar[j] })
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func rSortFloat64(ar []float64) {
	sort.Float64s(ar)
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func rSortRune(ar []rune) {
	sort.Slice(ar, func(i, j int) bool { return ar[i] < ar[j] })
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func fillInt(ar []int, x int) {
	for i := range ar {
		ar[i] = x
	}
}

func fillInt64(ar []int64, x int64) {
	for i := range ar {
		ar[i] = x
	}
}

func fillBool(ar []bool, x bool) {
	for i := range ar {
		ar[i] = x
	}
}

func fillFloat64(ar []float64, x float64) {
	for i := range ar {
		ar[i] = x
	}
}

func fillIntMatrix(ar [][]int, x int) {
	for i := range ar {
		fillInt(ar[i], x)
	}
}

func fillInt64Matrix(ar [][]int64, x int64) {
	for i := range ar {
		fillInt64(ar[i], x)
	}
}

func fillFloat64Matrix(ar [][]float64, x float64) {
	for i := range ar {
		fillFloat64(ar[i], x)
	}
}

func fillBoolMatrix(ar [][]bool, x bool) {
	for i := range ar {
		fillBool(ar[i], x)
	}
}

func plus(x, y int64) int64 {
	res := (x + y) % mod
	if res < 0 {
		res += mod
	}
	return res
}

func sub(x, y int64) int64 {
	res := (x - y) % mod
	if res < 0 {
		res += mod
	}
	return res
}

func mul(x, y int64) int64 {
	res := (x * y) % mod
	if res < 0 {
		res += mod
	}
	return res
}

func div(x, y int64) int64 {
	res := x * pow(y, mod-2) % mod
	if res < 0 {
		res += mod
	}
	return res
}

func pow(x, y int64) int64 {
	if y < 0 {
		return 0
	}
	if y == 0 {
		return 1
	}
	if y%2 == 1 {
		return (x * pow(x, y-1)) % mod
	}
	root := pow(x, y/2)
	return root * root % mod
}

func ns() string {
	line, _ := reader.ReadString('\n')
	return strings.TrimSpace(line)
}

func ni() int {
	line, _ := reader.ReadString('\n')
	i, _ := strconv.Atoi(strings.TrimSpace(line))
	return i
}

func nl() int64 {
	line, _ := reader.ReadString('\n')
	i, _ := strconv.ParseInt(strings.TrimSpace(line), 10, 64)
	return i
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/comment/direct/Java/C++/s456888268_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
#include <string>

using namespace std;

const long long MOD = 1e9 + 7;
const int dx[] = { -1, 0, 1, 0 };
const int dy[] = { 0, -1, 0, 1 };
const int dx8[] = { -1, -1, -1, 0, 0, 1, 1, 1 };
const int dy8[] = { -1, 0, 1, -1, 1, -1, 0, 1 };
const int INF = numeric_limits<int>::max() / 2;
const long long LINF = numeric_limits<long long>::max() / 3;
const double DINFINITY = numeric_limits<double>::max() / 3;
const double EPS = 1e-10;
const double PI = acos(-1);

void reverse(vector<int>& ar) {
    int len = ar.size();
    for (int i = 0; i < len / 2; i++) {
        swap(ar[i], ar[len - 1 - i]);
    }
}

void reverse(vector<long long>& ar) {
    int len = ar.size();
    for (int i = 0; i < len / 2; i++) {
        swap(ar[i], ar[len - 1 - i]);
    }
}

void reverse(vector<double>& ar) {
    int len = ar.size();
    for (int i = 0; i < len / 2; i++) {
        swap(ar[i], ar[len - 1 - i]);
    }
}

void reverse(string& s) {
    reverse(s.begin(), s.end());
}

template<typename T>
void reverse(vector<T>& ls) {
    int sz = ls.size();
    for (int i = 0; i < sz / 2; i++) {
        swap(ls[i], ls[sz - 1 - i]);
    }
}

template<typename T>
void reverse(T ar[], int size) {
    for (int i = 0; i < size / 2; i++) {
        swap(ar[i], ar[size - 1 - i]);
    }
}

int lowerBound(const vector<int>& a, int x) {
    return lower_bound(a.begin(), a.end(), x) - a.begin();
}

int upperBound(const vector<int>& a, int x) {
    return upper_bound(a.begin(), a.end(), x) - a.begin();
}

int rlowerBound(const vector<int>& a, int x) {
    return lower_bound(a.rbegin(), a.rend(), x) - a.rbegin();
}

int rupperBound(const vector<int>& a, int x) {
    return upper_bound(a.rbegin(), a.rend(), x) - a.rbegin();
}

int lowerBound(const vector<long long>& a, long long x) {
    return lower_bound(a.begin(), a.end(), x) - a.begin();
}

int upperBound(const vector<long long>& a, long long x) {
    return upper_bound(a.begin(), a.end(), x) - a.begin();
}

int rlowerBound(const vector<long long>& a, long long x) {
    return lower_bound(a.rbegin(), a.rend(), x) - a.rbegin();
}

int rupperBound(const vector<long long>& a, long long x) {
    return upper_bound(a.rbegin(), a.rend(), x) - a.rbegin();
}

int lowerBound(const vector<double>& a, double x) {
    return lower_bound(a.begin(), a.end(), x) - a.begin();
}

int upperBound(const vector<double>& a, double x) {
    return upper_bound(a.begin(), a.end(), x) - a.begin();
}

int rlowerBound(const vector<double>& a, double x) {
    return lower_bound(a.rbegin(), a.rend(), x) - a.rbegin();
}

int rupperBound(const vector<double>& a, double x) {
    return upper_bound(a.rbegin(), a.rend(), x) - a.rbegin();
}

int lowerBound(const string& a, char x) {
    return lower_bound(a.begin(), a.end(), x) - a.begin();
}

int upperBound(const string& a, char x) {
    return upper_bound(a.begin(), a.end(), x) - a.begin();
}

int rlowerBound(const string& a, char x) {
    return lower_bound(a.rbegin(), a.rend(), x) - a.rbegin();
}

int rupperBound(const string& a, char x) {
    return upper_bound(a.rbegin(), a.rend(), x) - a.rbegin();
}

template<typename T>
int lowerBound(const vector<T>& ls, const T& x) {
    return lower_bound(ls.begin(), ls.end(), x) - ls.begin();
}

template<typename T>
int upperBound(const vector<T>& ls, const T& x) {
    return upper_bound(ls.begin(), ls.end(), x) - ls.begin();
}

template<typename T>
int rlowerBound(const vector<T>& ls, const T& x) {
    return lower_bound(ls.rbegin(), ls.rend(), x) - ls.rbegin();
}

template<typename T>
int rupperBound(const vector<T>& ls, const T& x) {
    return upper_bound(ls.rbegin(), ls.rend(), x) - ls.rbegin();
}

vector<int> concat(int x, const vector<int>& arr) {
    vector<int> ret(arr);
    ret.insert(ret.begin(), x);
    return ret;
}

vector<int> concat(const vector<int>& arr, int x) {
    vector<int> ret(arr);
    ret.push_back(x);
    return ret;
}

vector<long long> concat(long long x, const vector<long long>& arr) {
    vector<long long> ret(arr);
    ret.insert(ret.begin(), x);
    return ret;
}

vector<long long> concat(const vector<long long>& arr, long long x) {
    vector<long long> ret(arr);
    ret.push_back(x);
    return ret;
}

int max(int x, int y) {
    return std::max(x, y);
}

int min(int x, int y) {
    return std::min(x, y);
}

int max(int x, int y, int z) {
    return std::max({x, y, z});
}

int min(int x, int y, int z) {
    return std::min({x, y, z});
}

long long max(long long x, long long y) {
    return std::max(x, y);
}

long long min(long long x, long long y) {
    return std::min(x, y);
}

long long max(long long x, long long y, long long z) {
    return std::max({x, y, z});
}

long long min(long long x, long long y, long long z) {
    return std::min({x, y, z});
}

double max(double x, double y) {
    return std::max(x, y);
}

double min(double x, double y) {
    return std::min(x, y);
}

double max(double x, double y, double z) {
    return std::max({x, y, z});
}

double min(double x, double y, double z) {
    return std::min({x, y, z});
}

void sort(vector<int>& ar) {
    std::sort(ar.begin(), ar.end());
}

void sort(vector<long long>& ar) {
    std::sort(ar.begin(), ar.end());
}

void sort(vector<double>& ar) {
    std::sort(ar.begin(), ar.end());
}

void sort(string& ar) {
    std::sort(ar.begin(), ar.end());
}

void rsort(vector<int>& ar) {
    std::sort(ar.begin(), ar.end());
    reverse(ar);
}

void rsort(vector<long long>& ar) {
    std::sort(ar.begin(), ar.end());
    reverse(ar);
}

void rsort(vector<double>& ar) {
    std::sort(ar.begin(), ar.end());
    reverse(ar);
}

void rsort(string& ar) {
    std::sort(ar.begin(), ar.end());
    reverse(ar);
}

void fill(vector<int>& arr, int x) {
    std::fill(arr.begin(), arr.end(), x);
}

void fill(vector<long long>& arr, long long x) {
    std::fill(arr.begin(), arr.end(), x);
}

void fill(vector<bool>& arr, bool x) {
    std::fill(arr.begin(), arr.end(), x);
}

void fill(vector<double>& arr, double x) {
    std::fill(arr.begin(), arr.end(), x);
}

void fill(vector<vector<int>>& arr, int x) {
    for (auto& row : arr) {
        std::fill(row.begin(), row.end(), x);
    }
}

void fill(vector<vector<long long>>& arr, long long x) {
    for (auto& row : arr) {
        std::fill(row.begin(), row.end(), x);
    }
}

void fill(vector<vector<double>>& arr, double x) {
    for (auto& row : arr) {
        std::fill(row.begin(), row.end(), x);
    }
}

void fill(vector<vector<bool>>& arr, bool x) {
    for (auto& row : arr) {
        std::fill(row.begin(), row.end(), x);
    }
}

long long plus(long long x, long long y) {
    long long res = (x + y) % MOD;
    return res < 0 ? res + MOD : res;
}

long long sub(long long x, long long y) {
    long long res = (x - y) % MOD;
    return res < 0 ? res + MOD : res;
}

long long mul(long long x, long long y) {
    long long res = (x * y) % MOD;
    return res < 0 ? res + MOD : res;
}

long long div(long long x, long long y) {
    long long res = x * pow(y, MOD - 2) % MOD;
    return res < 0 ? res + MOD : res;
}

long long pow(long long x, long long y) {
    if (y < 0) return 0;
    if (y == 0) return 1;
    if (y % 2 == 1) return (x * pow(x, y - 1)) % MOD;
    long long root = pow(x, y / 2);
    return root * root % MOD;
}

void solve() {
    string tmp;
    cin >> tmp;
    vector<char> s(tmp.begin(), tmp.end());
    int n = s.size();
    int ans = 0;
    int cnt = 0;
    for (int i = 0; i < n; i++) {
        if (s[i] == 'S') {
            cnt++;
        } else {
            if (cnt == 0) continue;
            ans++;
            cnt--;
        }
    }
    cout << n - 2 * ans << endl;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    solve();
    return 0;
}

// End of Code
